c++/src/kj/compat/http.c++

```diff
@@ -5709,17 +5709,34 @@ public:
       }).attach(kj::mv(address));
     }).split();
 
+    auto connection = kj::refcountedWrapper(kj::newPromisedStream(kj::mv(kj::get<1>(split))));
+
+    // XXX: I'm uncertain whether this is the best approach.
+    kj::Own<kj::AsyncIoStream> ref1 = connection->addWrappedRef();
```

Comment:
I'm uncertain about lifetimes here, but I need the tlsStarter closure to capture a reference to the AsyncIoStream so that it can upgrade it.